host/xtest/regression_1000.c

```diff
@@ -1473,3 +1476,266 @@ static void xtest_tee_test_1020(ADBG_Case_t *c)
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_NONE, TEEC_NONE, TEEC_NONE);
+
+	res = TEEC_InvokeCommand(&cs->session, TA_SIMS_OPEN_TA_SESSION,
+				 &op, &ret_orig);
+	if (res != TEEC_SUCCESS)
+		return TEEC_ERROR_GENERIC;
+
+	return res;
+}
+
+static TEEC_Result sims_get_counter(struct xtest_session *cs,
+				  uint32_t *counter)
+{
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig = 0;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_VALUE_OUTPUT,
+					 TEEC_NONE, TEEC_NONE, TEEC_NONE);
+
+	res = TEEC_InvokeCommand(&cs->session,
+				 TA_SIMS_CMD_GET_COUNTER, &op,
+				 &ret_orig);
+	*counter = op.params[0].value.a;
+
+	return res;
+}
+
+static TEEC_Result trigger_panic(struct xtest_session *cs, const TEEC_UUID *uuid)
+{
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig = 0;
+
+	if (!uuid) {
+		op.params[0].tmpref.buffer = (void *)NULL;
+		op.params[0].tmpref.size = 0;
+	} else {
+		op.params[0].tmpref.buffer = (void *)uuid;
+		op.params[0].tmpref.size = sizeof(TEEC_UUID);
+	}
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_NONE, TEEC_NONE, TEEC_NONE);
+
+	return TEEC_InvokeCommand(&cs->session, TA_SIMS_CMD_PANIC,
+				  &op, &ret_orig);
+}
+
+static void close_nsec_session(struct xtest_session *cs)
+{
+	TEEC_CloseSession(&cs->session);
+	TEEC_FinalizeContext(&cs->context);
+}
+
+static void test_panic_ca_to_ta(ADBG_Case_t *c, const TEEC_UUID *uuid,
+			       bool multi_instance)
+{
+	TEEC_Result exp_res = TEEC_ERROR_GENERIC;
+	uint32_t counter = 0;
+	uint32_t exp_counter = 0;
+	struct xtest_session cs[3];
+
+	memset(cs, 0, sizeof(cs));
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, open_nsec_session(&cs[0], uuid)))
+		return;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, open_nsec_session(&cs[1], uuid)))
+		goto bail0;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
+				     sims_get_counter(&cs[0], &counter)))
+		goto bail1;
+
+	if (!ADBG_EXPECT(c, 0, counter))
+		goto bail1;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, 0, sims_get_counter(&cs[1], &counter)))
+		goto bail1;
+
+	exp_counter = multi_instance ? 0 : 1;
+	if (ADBG_EXPECT(c, exp_counter, counter))
+		goto bail1;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEE_ERROR_TARGET_DEAD,
+				     trigger_panic(&cs[1], NULL)))
+		goto bail1;
+
+	exp_res = multi_instance ? TEEC_SUCCESS : TEE_ERROR_TARGET_DEAD;
+	if (!ADBG_EXPECT_TEEC_RESULT(c, exp_res,
+				     sims_get_counter(&cs[0], &counter)))
+		goto bail1;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEE_ERROR_TARGET_DEAD,
+				     sims_get_counter(&cs[1], &counter)))
+		goto bail1;
+
+	/* Attempt to open a session on panicked context */
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEE_ERROR_TARGET_DEAD,
+				     open_nsec_session(&cs[1], uuid)))
+		goto bail1;
+
+	/* Open a new session on a new context */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, open_nsec_session(&cs[2], uuid)))
+		goto bail1;
+
+	/* Optional test, already tested above */
```

Comment:
I still think this test is useless. Or maybe I feel uncomfortable with this inline comment.
Maybe keep this test and change comment to `/* Sanity check of still valid TA context */ `.

Reply:
The comment, I agree now when I red it again, is at least unfortunate. The test itself could be useful when showing that other context is available for new connections.

Reply:
Could you change the inline comment into `/* Sanity check of still valid TA context */`?